src/java.base/share/classes/java/lang/runtime/Carrier.java

```diff
@@ -353,64 +357,62 @@ private static MethodHandle reshapeComponent(CarrierShape carrierShape,
             /**
              * Constructor.
              *
-             * @param intCount     slot count required for longs and integers
-             * @param objectCount  slot count required for objects
+             * @param primitiveCount  slot count required for primitives
+             * @param objectCount     slot count required for objects
              */
-            CarrierArray(int intCount, int objectCount) {
-                this.integers = new int[intCount];
+            CarrierArray(int primitiveCount, int objectCount) {
+                this.primitives = new long[primitiveCount];
                 this.objects = new Object[objectCount];
             }
 
             /**
-             * Check index and compute offset for unsafe long access.
+             * Check index and compute offset for unsafe access.
              *
-             * @param i  index in int[]
+             * @param i  index in primitive[]
              *
-             * @return offset for unsafe long access
+             * @return offset for unsafe access
              */
-            private long longOffset(int i) {
-                if (i < 0 || integers.length <= i) {
-                    throw new RuntimeException("long index out of range: " + i);
+            private long offset(int i) {
+                if (i < 0 || primitives.length <= i) {
+                    throw new RuntimeException("primitive index out of range: " + i);
                 }
 
-                return Unsafe.ARRAY_INT_BASE_OFFSET +
+                return Unsafe.ARRAY_LONG_BASE_OFFSET +
```

Comment:
Shouldn't you add the offset of the first `int` value in the array to the resulting expression? E.g. I'm assuming we're storing `long`s first and `int`s after? If so, `int` values will be located at an offset from the start of the array.

Reply:
Ah ok - I see - the constructor method handle binds the correct offset, adding the required constant offset (for all the long values). So my comment it's mostly stylistic.

Reply:
I'll add commentary.